vitest-setup.ts

```diff
@@ -1,17 +1,31 @@
 import { vi } from "vitest";
+import { EnvironmentMode } from "./src/config";
 
-vi.mock("./src/config", async () => ({
-  AWS_REGION: "ca-central-1",
-  SERVER_PORT: 3001,
-  LOCALSTACK_ENDPOINT: undefined,
-  ZITADEL_APPLICATION_KEY: JSON.stringify({
-    keyId: "test",
-    clientId: "test",
-    key: "test",
-  }),
-  ZITADEL_DOMAIN: "test",
-}));
+vi.doMock("./src/config", async (importOriginal) => {
```

Comment:
Using `doMock` instead of `mock` to be able to use the `EnvironmentMode` enum type in the mock implementation.